resources/js/pages/AdminBugPanel.vue

```diff
@@ -0,0 +1,140 @@
+        <div>
+            <h4>{{ "this is an admin bug panel" }}</h4>
+        </div>
+        <div>
+            <p>
+                sort by:
+                <template v-for="sortable in sortables">
+                    <button v-if="sortable != 'type'" :key="sortable" v-on:click="sort(sortable)">{{sortable}}</button>
+                    <button v-if="sortable == 'type'" :key="sortable" v-on:click="sort(sortable)">{{sortable}}: {{currentSortType}}</button>
+                </template>
+                (click again to reverse order)
+            </p>
+        </div>
+        <template v-for="bug in sortedBugs">
+            <div :key="bug.id" :title="bug.title" class="bug">
+                <div :class="headerColour(bug.severity) + ' d-flex header'">
+                    <span>{{bug.title}}</span>
+                    <span class="m-auto">{{bug.type}}</span>
+                    <span class="ml-auto">{{bug.severity}}</span>
+                </div>
+                <div class="bug-content">
+                    <p>Page: {{bug.page}}</p>
+                    <p v-if="bug.image_link">Image: {{bug.image_link}}</p>
+                    <p>Comment: "{{bug.comment}}"</p>
+                    <p>Reported by user: {{bug.user_id}} </p>
+                    <p>Status: {{bug.status}}</p>
+                    <p>time: {{bug.time_created}}</p>
+                    <span v-if="bug.admin_comment">Admin comment: "{{bug.admin_comment}}"</span>
+                </div>
+            </div>
+        </template>
+        debug: currentSort: {{currentSort}} | currentSortDir: {{currentSortDir}} | currentSortType: {{currentSortType}}
+    </div>
+</template>
+
+
+<script>
+import {mapGetters} from 'vuex';
+export default {
+    mounted() {
+        this.$store.dispatch('admin/checkAdmin');
+        this.$store.dispatch('bugReport/fetchBugs');
+    },
+    computed: {
+        ...mapGetters({
+            bugs: 'bugReport/getBugs',
+        }),        
+        sortedBugs:function() {
+            return this.bugs.sort((a,b) => {
+                let modifier = 1;
+                if(this.currentSort === 'type') {
+                    modifier = (this.types.length - this.types.indexOf(this.currentSortType));
+                    let tempA = (this.types.indexOf(a[this.currentSort]) + modifier) % this.types.length;
+                    let tempB = (this.types.indexOf(b[this.currentSort]) + modifier) % this.types.length;
+                    if(tempA < tempB) return -1;
+                    if(tempA > tempB) return 1;
+                } else {
+                    if(this.currentSortDir === 'desc') modifier = -1;
+                    if(a[this.currentSort] < b[this.currentSort]) return -1 * modifier;
+                    if(a[this.currentSort] > b[this.currentSort]) return 1 * modifier;
+                }
+                return 0;
+            })
+        },
+        //this may be needed for language support
+        sortedTypes:function() {
```

Comment:
Don't need `:function`, the parenthesis make it a function. Same up here on sortedBugs